addon/services/file-manager.js

```diff
@@ -1,133 +1,266 @@
 import Ember from 'ember';
 
+/**
+ * An Ember service for doing things to files.
+ * Essentially a wrapper for the Waterbutler API.
+ * http://waterbutler.readthedocs.io/
+ *
+ * @class file-manager
+ * @extends Ember.Service
+ */
+
 export default Ember.Service.extend({
     session: Ember.inject.service(),
     store: Ember.inject.service(),
 
-    // TODO: After performing a Waterbutler action, it takes a little while
-    // (maybe a minute, depending on caching options) for the OSF API to catch
-    // up and start serving the updated metadata.
-    //
-    // For actions on existing entities, this is fine, and file-manager
-    // uses info from WB to update the Ember store so everything looks right
-    // to the user, right away.
-    //
-    // For actions that create entities (addSubfolder, uploadFile, copy),
-    // this is a problem, because we need the OSF info (guid, WB links)
-    // before letting the user do anything to their new file. At the moment,
-    // after you create a file, you should go get a cup of coffee, come back,
-    // and hard reload.
-
-    ///////////////////// File-only actions /////////////////////
-
-    getContents(file) {
+    isReloading: false,
```

Comment:
I like the hash idea, but I don't think we can use `buildUrl` -- I originally put the bypasscache thing there, but it didn't work for reloading the list of files in a folder (`folder.get('files').reload()`), because it replaces the built URL with the relationship link.

I assume there's a way to get the `files` relationship link from a given `file` object, file-manager could have a whitelist of urls that should bypass the cache. Hackier than using the file id, but better than `isReloading`